include/mbedtls/mps/reader.h

```diff
@@ -12,21 +12,28 @@
  * a 'consumer' which fetches and processes it in chunks of
  * again arbitrary, and potentially different, size.
  *
- * The reader abstracts away the underlying pointer arithmetic
- * as well as the merging of incoming data chunks in case the
- * consumer requests data in larger chunks than what the
- * producer provides.
- *
- * The basic flow of operation is that the producer hands
- * an incoming data buffer to the reader, switching from
- * 'producing' to 'consuming' mode. The consumer subsequently
- * fetches and processes the buffer content. Once that's done --
- * or partially done and a consumer's requests can't be fulfilled --
- * the producer revokes the reader's access to the incoming
- * data buffer, putting the reader back to producing mode;
- * the producer subsequently gathers more incoming data and hands
- * it to reader until the latter switches back to consuming mode
- * if enough data is available.
+ * Readers can be seen as datagram-to-stream converters,
+ * and they abstract away the following two tasks from the user:
+ * 1. The pointer arithmetic of stepping through a producer-
+ *    provided chunk in smaller chunks.
+ * 2. The merging of incoming data chunks in case the
+ *    consumer requests data in larger chunks than what the
+ *    producer provides.
+ *
+ * The basic abstract flow of operation is the following:
+ * - Initially, the reader is in 'producing mode'.
+ * - The producer hands an incoming data buffer to the reader,
+ *   moving it from 'producing' to 'consuming' mode.
+ * - The consumer subsequently fetches and processes the buffer
+ *   content. Once that's done -- or partially done and a consumer's
```

Comment:
I'd either omit the "a" - "and consumer's requests", or use the singular form - "a consumer's request"

Reply:
Thanks, it should be `consumer's request`.